src/main.rs

```diff
@@ -432,44 +460,64 @@ fn render_table(f: &mut Frame, app: &mut App, area: Rect) {
 }
 
 fn render_footer(f: &mut Frame, app: &mut App, area: Rect) {
-    let footer_text = match app.mode {
+    let info_text = match app.mode {
         Mode::Table => TABLE_FOOTER,
         Mode::Graph => GRAPHS_FOOTER,
         Mode::Filter => FILTER_FOOTER,
+        Mode::Sort => SORT_INFO_FOOTER,
     };
-    let info_footer = Paragraph::new(Line::from(footer_text)).centered().block(
+    let info_footer = Paragraph::new(Line::from(info_text)).centered().block(
         Block::default()
             .borders(Borders::ALL)
             .border_type(BorderType::Double),
     );
 
-    match app.mode {
-        Mode::Table | Mode::Graph => {
-            f.render_widget(info_footer, area);
-        }
-        Mode::Filter => {
-            let filter_area = Layout::default()
-                .direction(Direction::Horizontal)
-                .constraints([Constraint::Percentage(75), Constraint::Percentage(25)].as_ref())
-                .split(area);
-
-            let filter_input = app.filter_input.lock().unwrap();
-            let filter_footer = Paragraph::new(filter_input.value()).block(
-                Block::default()
-                    .padding(Padding::horizontal(1))
-                    .borders(Borders::ALL)
-                    .border_type(BorderType::Double)
-                    .title(" Filter Name/Type "),
-            );
-
-            f.render_widget(filter_footer, filter_area[0]); // Left filter footer
-            f.render_widget(info_footer, filter_area[1]); // Right info footer
-
-            // Displays cursor when inputting
-            f.set_cursor(
-                filter_area[0].x + filter_input.visual_cursor() as u16 + 2,
-                filter_area[0].y + 1,
-            )
+    if let Mode::Table | Mode::Graph = app.mode {
```

Comment:
This line confuses me, what exactly is it doing?

Reply:
This is another way to doing enum pattern matching, similar to `if app.mode == Mode::Table || app.mode == Mode::Graph`, or:
```rust
match app.mode {
    Mode::Table | Mode::Graph => ...,
```

I made a slight modification to this part so that it's less "nested", if it helps improve the readability.